h2/src/main/org/h2/pagestore/PageInputStream.java

```diff
@@ -101,8 +101,7 @@ private void fillBuffer() {
                     endOfFile = true;
                     return;
                 }
-            }
-            if (trunk != null) {
+            } else {
```

Comment:
no, that's not right, because we assign to trunk inside the if block

Reply:
Oops. I agree.